<?xml version="1.0" ?>
<launch>
	<arg name="namespace" default="sam"/>

  <!-- Frames -->
  <arg name="map_frame" default="map"/>
  <arg name="odom_frame"  default="$(arg namespace)/odom"/>
  <arg name="base_frame"  default="sam_test"/>

  <!-- PF args -->
  <arg name="particle_count"  default="50"/>

  <!-- [x, y, z, roll, pitch, yaw] -->
  <arg name="init_covariance" default="[0.01, 0.01, 0.0, 0.0, 0.0, 0.0]"/>
  <arg name="motion_covariance" default="[0.0000, 0.0000, 0.0, 0.0, 0.0, 0.00000001]"/>
  <arg name="resampling_noise_covariance" default="[0.1, 0.1, 0.0, 0.0, 0.0, 0.001]"/>
  <arg name="measurement_std"  default="1."/>

  <group ns="$(arg namespace)">
    <node type="auv_pf.py" pkg="auv_particle_filter" name="auv_pf" output="screen">
      <param name="particle_count"          type= "int"     value="$(arg particle_count)" />
      <param name="init_covariance"                         value="$(arg init_covariance)" />
      <param name="resampling_noise_covariance"             value="$(arg resampling_noise_covariance)" />
      <param name="measurement_std"                         value="$(arg measurement_std)" />
      <param name="motion_covariance"                       value="$(arg motion_covariance)" />
      <param name="odometry_topic"                          value="/$(arg namespace)/dr/dvl_dr"/>
      <param name="gps_odom_topic"                          value="/sam/dr/gps_odom_sam"/>
      <param name="map_frame"                               value="$(arg map_frame)" />
      <param name="odom_frame"                              value="$(arg odom_frame)" />
      <param name="base_link"                               value="$(arg namespace)/base_link" />
      <param name="average_pose_topic"                      value="/$(arg namespace)/avg_pose"/>
      <param name="particle_poses_topic"                    value="/$(arg namespace)/particle_poses"/>
    </node>
  </group>

</launch>

